File: backend/src/main/java/de/grimsi/gameyfin/mapper/PlayerPerspectiveMapper.java
package de.grimsi.gameyfin.mapper;

import com.igdb.proto.Igdb;
import de.grimsi.gameyfin.entities.PlayerPerspective;

import java.util.List;

public class PlayerPerspectiveMapper {    
    
    public static PlayerPerspective toPlayerPerspective(Igdb.PlayerPerspective g) {
    return PlayerPerspective.builder()
            .slug(g.getSlug())
            .name(g.getName())
            .build();
}

    public static List<PlayerPerspective> toPlayerPerspectives(List<Igdb.PlayerPerspective> g) {
        return g.stream().map(PlayerPerspectiveMapper::toPlayerPerspective).toList();
    }
}

File: plugins/igdb/src/main/kotlin/org/gameyfin/plugins/metadata/igdb/mapper/PlayerPerspectiveMapper.kt
package org.gameyfin.plugins.metadata.igdb.mapper

import org.gameyfin.pluginapi.gamemetadata.PlayerPerspective
import org.slf4j.LoggerFactory

class PlayerPerspectiveMapper {
    companion object {
        private val log = LoggerFactory.getLogger(this::class.java)

        fun playerPerspective(perspective: proto.PlayerPerspective): PlayerPerspective {
            return when (perspective.slug) {
                "first-person" -> PlayerPerspective.FIRST_PERSON
                "third-person" -> PlayerPerspective.THIRD_PERSON
                "bird-view-isometric" -> PlayerPerspective.BIRD_VIEW_ISOMETRIC
                "bird-view-slash-isometric" -> PlayerPerspective.BIRD_VIEW_ISOMETRIC
                "side-view" -> PlayerPerspective.SIDE_VIEW
                "text" -> PlayerPerspective.TEXT
                "auditory" -> PlayerPerspective.AUDITORY
                "virtual-reality" -> PlayerPerspective.VIRTUAL_REALITY
                else -> {
                    log.warn("Unknown player perspective: {}", perspective.slug)
                    PlayerPerspective.UNKNOWN
                }
            }
        }
    }
}